Stripe/PublicHeaders/STPCardValidator.h

```diff
@@ -0,0 +1,89 @@
+/**
+ *  The card brand for a card number or substring thereof.
+ *
+ *  @param cardNumber A card number, or partial card number. For example, @"4242", @"5555555555554444", or @"123".
+ *
+ *  @return The brand for that card number. The example parameters would return STPCardBrandVisa, STPCardBrandMasterCard, and STPCardBrandUnknown, respectively.
+ */
++ (STPCardBrand)brandForNumber:(nonnull NSString *)cardNumber;
+
+/**
+ *  The number length for cards associated with a card brand. For example, Visa card numbers contain 16 characters, while American Express cards contain 15 characters.
+ */
++ (NSInteger)lengthForCardBrand:(STPCardBrand)brand;
+
+/**
+ *  The length of the final grouping of digits to use when formatting a card number for display. For example, Visa cards display their final 4 numbers, e.g. "4242", while American Express cards display their final 5 digits, e.g. "10005".
+ */
++ (NSInteger)fragmentLengthForCardBrand:(STPCardBrand)brand;
+
+/**
+ *  Validates an expiration month, passed as an (optionally 0-padded) string. Example valid values are "3", "12", and "08". Example invalid values are "99", "a", and "00". Incomplete values include "0" and "1".
+ *
+ *  @param expirationMonth A string representing a 2-digit expiration month for a payment card.
+ *
+ *  @return STPCardValidationStateValid if the month is valid, STPCardValidationStateInvalid if the month is invalid, or STPCardValidationStateIncomplete if the month is a substring of a valid month (e.g. @"0" or @"1").
+ */
++ (STPCardValidationState)validationStateForExpirationMonth:(nonnull NSString *)expirationMonth;
+
+/**
+ *  Validates an expiration year, passed as a string representing the final 2 digits of the year. This considers the period between the current year until 2099 as valid times. An example valid value would be "16" (assuming the current year, as determined by [NSDate date], is 2015). Will return STPCardValidationStateInvalid for a month/year combination that is earlier than the current date (i.e. @"15" and @"04" in October 2015. Example invalid values are "00", "a", and "13". Any 1-digit string will return STPCardValidationStateIncomplete.
+ *
+ *  @param expirationYear A string representing a 2-digit expiration year for a payment card.
+ *  @param expirationMonth A string representing a 2-digit expiration month for a payment card. See -validationStateForExpirationMonth for the desired formatting of this string.
+ *
+ *  @return STPCardValidationStateValid if the year is valid, STPCardValidationStateInvalid if the year is invalid, or STPCardValidationStateIncomplete if the year is a substring of a valid year (e.g. @"1" or @"2").
+ */
++ (STPCardValidationState)validationStateForExpirationYear:(nonnull NSString *)expirationYear inMonth:(nonnull NSString *)expirationMonth;
+
+/**
+ *  The max CVC length for a card brand (for context, American Express CVCs are 4 digits, while all others are 3).
+ */
++ (NSUInteger)maxCvcLengthForCardBrand:(STPCardBrand)brand;
```

Comment:
nit: `maxCVCLengthForCardBrand` would be more consistent with `validationStateForCVC`